### PR TITLE
Fix pkgconfig for gnutls

### PR DESCRIPTION
**Pull request by tuxmaster5000**
Fix https://github.com/greenbone/openvas-smb/issues/29

### CODE REVIEW

**Comment by jjnicola** [comment]
Hi @tuxmaster5000,
Does it work for you? It does not pass the CI tests. I have tested this PR in my local dev environment and it is broken as well.

**Comment by tuxmaster5000** [comment]
Yes,it will work on my side, but I use CentOS as the build OS.  Can you modify the test, so that the whole compile call are visible?
On my build at that stage there is no '-R'  option used:
```
[ 91%] Linking C executable winexe
cd /builddir/build/BUILD/openvas-smb-1.0.5/winexe && /usr/bin/cmake3 -E cmake_link_script CMakeFiles/winexe.dir/link.txt --verbose=1
/usr/lib64/ccache/cc -Wno-unused-result -fno-strict-aliasing -fPIC -O2 -D_FORTIFY_SOURCE=2 -fstack-protector -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1  -m64 -mtune=generic  -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -rdynamic CMakeFiles/winexe.dir/winexe.c.o CMakeFiles/winexe.dir/service.c.o CMakeFiles/winexe.dir/async.c.o CMakeFiles/winexe.dir/winexesvc_exe.c.o  -o winexe  ../samba/libsamba-static.a -ldl -lgnutls -lpopt -Wl,-z,relro -Wl,-z,now -L/usr/lib64/heimdal -lkrb5 -lgssapi -lasn1 -lcom_err -pthread -L/usr/lib64/gvm-helper -lgnutls -ltspi -latomic -lhogweed -lgmp -ltasn1 -lidn2 -lunistring -lnettle -lhdb 
```

**Comment by jjnicola** [comment]
We will update the image used for running the tests, as it is currently based on stretch, but it should be buster.
It finally worked in my local env, I just was missing a dependency for gnutls version I have installed in my Debian. I have not seen the "-R" issue either.

File: samba/CMakeLists.txt [hunk 1]
```diff
@@ -450,5 +451,5 @@ add_custom_command(
 add_library(samba-static STATIC ${RPC_GEN_SRC} ${SAMBA_SRC} ${RPC_GEN_HEADERS})
 # Link against static heimdal-gssapi ldflags returned by pkg-config
 # Link against libhdb as well since it is missing from the list
-target_link_libraries (samba-static ${GSSAPI_STATIC_LDFLAGS} hdb)
+target_link_libraries (samba-static ${GSSAPI_STATIC_LDFLAGS} ${GNUTLS_STATIC_LDFLAGS} hdb )
```
Comment (cfi-gb) [inline samba/CMakeLists.txt:454]:
```suggestion
target_link_libraries (samba-static ${GSSAPI_STATIC_LDFLAGS} ${GNUTLS_STATIC_LDFLAGS} hdb)
```

I guess the space is superfluous and should be dropped to keep the coding style consistent?

**Comment by jjnicola** [comment]
Hi @tuxmaster5000,

could you add Changlog entry? after your commit, the test will run again, using the new docker images. This time, all check should pass without problems.

You can check the Changelog format [here](https://github.com/greenbone/openvas-scanner/blob/master/CHANGELOG.md)

**Comment by tuxmaster5000** [comment]
Now the change log entry was added.

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**